packages/backend-ai/src/conversation/factory.test.ts

```diff
@@ -0,0 +1,125 @@
+  app.node.setContext('amplify-backend-namespace', 'testBackendId');
+  app.node.setContext('amplify-backend-type', 'branch');
+  const stack = new Stack(app);
+  return stack;
+};
+
+void describe('ConversationHandlerFactory', () => {
+  let rootStack: Stack;
+  let getInstanceProps: ConstructFactoryGetInstanceProps;
+  let resourceNameValidator: ResourceNameValidator;
+
+  beforeEach(() => {
+    rootStack = createStackAndSetContext();
+
+    const constructContainer = new ConstructContainerStub(
+      new StackResolverStub(rootStack)
+    );
+
+    const outputStorageStrategy = new StackMetadataBackendOutputStorageStrategy(
+      rootStack
+    );
+
+    resourceNameValidator = new ResourceNameValidatorStub();
+
+    getInstanceProps = {
+      constructContainer,
+      outputStorageStrategy,
+      resourceNameValidator,
+    };
+  });
+
+  void it('creates singleton instance', () => {
+    const factory = defaultEntryHandler;
+    const instance1 = factory.getInstance(getInstanceProps);
+    const instance2 = factory.getInstance(getInstanceProps);
+    assert.strictEqual(instance1, instance2);
+  });
+
+  void it('resolves default entry when not specified', () => {
+    const factory = defaultEntryHandler;
+    const lambda = factory.getInstance(getInstanceProps);
+    const template = Template.fromStack(Stack.of(lambda.resources.lambda));
+    template.resourceCountIs('AWS::Lambda::Function', 1);
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      Handler: 'index.handler',
+    });
+  });
+
+  void it('uses entry that is explicitly specified', () => {
+    const factory = customEntryHandler;
+    const lambda = factory.getInstance(getInstanceProps);
+    const template = Template.fromStack(Stack.of(lambda.resources.lambda));
+    template.resourceCountIs('AWS::Lambda::Function', 1);
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      Handler: 'index.handler',
+    });
+  });
+
+  void it('throws on invalid name', () => {
+    mock
+      .method(resourceNameValidator, 'validate')
+      .mock.mockImplementationOnce(() => {
+        throw new Error('test validation error');
+      });
+    const factory = defineConversationHandlerFunction({
+      name: 'this!is@wrong',
+      models: [],
+    });
+    assert.throws(() => factory.getInstance(getInstanceProps), {
+      message: 'test validation error',
+    });
+  });
+
+  void describe('storeOutput', () => {
+    void it('stores output using the provided strategy', () => {
+      const factory = defineConversationHandlerFunction({
+        entry: './test-assets/with-default-entry/handler.ts',
+        name: 'testHandlerName',
+        models: [],
+      });
+      factory.getInstance(getInstanceProps);
+      const template = Template.fromStack(rootStack);
+      const outputValue =
+        template.findOutputs('definedFunctions').definedFunctions.Value;
+      assert.deepStrictEqual(outputValue, {
```

Comment:
assert.deepStrictEqual(outputValue, <something in the stack>.ref) ?

Reply:
That seems to be not doable.
```
      console.log(lambda.resources.lambda.functionName);
      console.log(lambda.resources.cfnResources.cfnFunction.ref);
```
prints
```
${Token[TOKEN.208]}
${Token[TOKEN.207]}
```

I also looked at APIs in `Template` but they don't return desired values either.

It seems this is all we can do here.
Similar places https://github.com/search?q=repo%3Aaws-amplify%2Famplify-backend+%27Fn%3A%3AGetAtt%27&type=code